packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -706,6 +708,44 @@ export function createRouteManifest(
 	};
 }
 
+async function getRoutePrerenderOption(
+	route: RouteData,
+	settings: AstroSettings,
+	fsMod: typeof nodeFs | undefined,
+	logger: Logger,
+) {
+	if (route.type !== 'page' && route.type !== 'endpoint') return;
+	const localFs = fsMod ?? nodeFs;
+	const content = localFs.readFileSync(
```

Comment:
I think it's worth trying to use `createReadStream` here to avoid reading the entire file unnecessarily. The code would be something like this (pseudo):

```js
let stream = fs.createReadStream(filepath);
let fm = 0;
for await(let chunk of stream) {
  if(/^---/.test(chunk)) {
    fm++;
    if(fm === 2) break;

     const match = /^\s*export\s+const\s+prerender\s*=\s*(true|false);?/m.exec(content);
     if (match) {
       route.prerender = match[1] === 'true';
       break;
     }
  }
}
```

Logic here is:

1. Read a chunk at a time.
2. Count the frontmatter ticks, if we hit a second one we don't need to keep reading.
3. If we find a prerender option we don't need to keep reading.

Reply:
I benchmarked this locally and checking for the frontmatter with the regex took more time than not checking